### DISCUSSION TITLE
Custom linter rules

### DISCUSSION THREAD

**Discussion by vakhramoff** [taiga-family/taiga-ui#897] [Ideas]
Hey, folks 👋🏻

I guess, we can write cool rules/plugins for eslint, which could help to prevent us from making some mistakes.

For example, when we use `TuiDestroyService`, we could forget to provide an individual instance in the providers array, which could lead to the catastrophic impact on our app, and it would be difficult to find a place where we forgot to write just one string.

So the idea💡 is to discuss whether we need such tools🧰?

**Comment by waterplea** [comment]
A rule world be to have @Self decorator in front of it.
